core/src/Kokkos_Core.hpp

```diff
@@ -115,6 +90,41 @@ struct InitArguments {
         disable_warnings{dw} {}
 };
 
+namespace Impl {
+
+/* ExecSpaceManager - Responsible for initializing all of the registered
+ * backends. Backends are registered using the register_space_initializer()
+ * function which should be called from a global context so that it is called
+ * prior to initialize_spaces() which is called from Kokkos::initialize()
+ */
+class ExecSpaceManager {
+  std::map<std::string, std::unique_ptr<ExecSpaceInitializerBase>>
+      exec_space_factory_list;
+
+ public:
+  ExecSpaceManager() = default;
+
+  void register_space_factory(std::string name,
+                              std::unique_ptr<ExecSpaceInitializerBase> ptr);
+  void initialize_spaces(const Kokkos::InitArguments& args);
+  void finalize_spaces(const bool all_spaces);
+  void static_fence();
+  void print_configuration(std::ostringstream& msg, const bool detail);
+  static ExecSpaceManager& get_instance();
+
+ protected:
+  static ExecSpaceManager g_space_initializer;
+};
+
+template <class SpaceInitializerType>
+int initialize_space_factory(std::string name) {
+  auto space_ptr = std::make_unique<SpaceInitializerType>();
+  ExecSpaceManager::get_instance().register_space_factory(name,
+                                                          std::move(space_ptr));
+  return 1;
+};
```

Comment:
```suggestion
}
```
to make `Travis` happy.